### PR TITLE
feat(Tag): new component

### LINKED ISSUE #240: [Feat]: New Tag component
**Description**:

The `Tag `component is used to display small, labeled elements, often used for categorization or tagging content. It is ideal for use in filtering systems, labels in item lists, or status indicators in dashboards or user profiles.

**Requirements**:

1. **Component Implementation:**

- Implement the `Tag `component in TypeScript.
- Utilize SCSS with BEM methodology for styling.
- Integrate design tokens from @geneui/tokens for consistent colors, font sizes, spacing, and border styles.
- Support different variants, such as filled, outlined, and ghost styles.
- Allow customization of colors, sizes, and icons within the tag.

2. **Unit Tests:**

- 
- Write unit tests using Jest and React Testing Library.
- Ensure test coverage for all props, including rendering different styles, sizes, and icons.
- Test for edge cases such as missing content, unsupported configurations, and large tag sizes.
- 

3. **Storybook Stories:**

- Create a Storybook story for the Tag component.
- Showcase different variants, sizes, and icon usages.
- Create controls for the props so users can interact with the Tag component in Storybook.
- Document all props of the public interfaces.

4. **Figma Design:**

Follow the design outlined in the provided[ Figma link.](https://www.figma.com/design/XmW63z2fXfHBC95eCM36ed/Gene-UI%3A-Components?node-id=141-60&p=f&m=dev)

**Checklist**:

- [ ]  Component is implemented in TypeScript.
- [ ]  Styling follows SCSS using BEM methodology.
- [ ]  Design tokens from @geneui/tokens are used for colors, font sizes, spacing, and borders.
- [ ]  Unit tests written and cover different states and edge cases.
- [ ]  Storybook stories created with examples of all props and configuration options.
- [ ]  Figma design matches the component implementation.

### PR DESCRIPTION
**Pull request by arthur-grigoryan-softconstruct**
### Chromatic Build Summary 
**Build URL:** https://www.chromatic.com/build?appId=66daee7859a843dd87bfb3d7&number=241 
**Storybook URL:** https://66daee7859a843dd87bfb3d7-sjeurzgkeb.chromatic.com/

### CODE REVIEW

File: src/components/molecules/Tag/Tag.test.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    });
+
+    it("renders withIcon prop correctly", () => {
+        expect(setup.find(".tag__icon").exists()).toBeTruthy();
+        const wrapper = setup.setProps({ withIcon: false });
+        expect(wrapper.find(".tag__icon").exists()).toBeFalsy();
+    });
+
+    it("handles close button's click", () => {
+        setup.find(".tag__button .button_icon_only").simulate("click");
```
Comment (GaroGabrielyan) [inline src/components/molecules/Tag/Tag.test.tsx:48]: Instead of using button local class, you can import and use Button component.

```suggestion
        setup.find(Button).simulate("click");
```

File: src/components/molecules/Tag/Tag.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import React, { FC } from "react";
+import classNames from "classnames";
+import { Close, TagOutline, WarningFill, ErrorAlertFill } from "@geneui/icons";
+
+// Styles
+import "./Tag.scss";
+
+// Components
+import Button from "../../atoms/Button/Button";
```
Comment (GaroGabrielyan) [inline src/components/molecules/Tag/Tag.tsx:9]: Please reorder imports, make the last import // Styles

File: src/components/molecules/Tag/Tag.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+interface ITagProps {
+    /**
+     * Tag content text
+     */
+    text: string;
+    /**
+     * Tag state <br/>
+     * Possible values: `rest | error | warning | disabled`
+     */
+    state?: "rest" | "error" | "warning" | "disabled";
```
[the remark this answers is not in the document]
Comment (arthur-grigoryan-softconstruct) [reply src/components/molecules/Tag/Tag.tsx:27]: sounds good

[the remark this answers is not in the document]
Comment (arthur-grigoryan-softconstruct) [reply src/components/molecules/Tag/Tag.tsx:27]: done

File: src/components/molecules/Tag/Tag.test.tsx (hunk 1 above)
Comment (arthur-grigoryan-softconstruct) [reply src/components/molecules/Tag/Tag.test.tsx:48]: Done

File: src/components/molecules/Tag/Tag.tsx (hunk 2 above)
Comment (arthur-grigoryan-softconstruct) [reply src/components/molecules/Tag/Tag.tsx:9]: Done

**Review by hamikhambardzumyan — CHANGES_REQUESTED** [review]

File: src/components/molecules/Tag/Tag.test.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import React from "react";
+import { ReactWrapper, mount } from "enzyme";
+
+// Components
+import Tag, { ITagProps } from "./index";
+import Button from "../../atoms/Button/Button";
```
[the remark this answers is not in the document]
Comment (arthur-grigoryan-softconstruct) [reply src/components/molecules/Tag/Tag.test.tsx:6]: done

File: src/components/molecules/Tag/Tag.test.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    it("renders content text correctly", () => {
+        expect(setup.find(".tag__text").contains(tagText)).toBeTruthy();
+    });
+
+    it.each<ITagProps["type"]>(["rest", "error", "warning"])("should have %s state", (type) => {
+        const wrapper = setup.setProps({ type });
+        expect(wrapper.find(`.tag_state_${type}`).exists()).toBeTruthy();
+    });
+
+    it("renders isDisabled prop correctly", () => {
```
[the remark this answers is not in the document]
Comment (arthur-grigoryan-softconstruct) [reply src/components/molecules/Tag/Tag.test.tsx:37]: done

File: src/components/molecules/Tag/Tag.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    withIcon = true,
+    onClose
+}) => {
+    const Icon = icons[type];
+    return (
+        <div
+            className={classNames(
+                "tag",
+                `tag_size_${size}`,
+                { [`tag_state_${type}`]: !isDisabled },
```
[the remark this answers is not in the document]
Comment (arthur-grigoryan-softconstruct) [reply src/components/molecules/Tag/Tag.tsx:69]: done

File: src/components/molecules/Tag/Tag.tsx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+/**
+ * Tag is used to label, categorize, and organize content within an interface. It can be used to highlight keywords, topics, or attributes related to an item. Tags enhance user navigation and search functionality by providing a quick way to filter and identify relevant information.
+ */
+const Tag: FC<ITagProps> = ({
+    className,
+    text,
+    type = "rest",
+    isDisabled,
```
[the remark this answers is not in the document]
Comment (arthur-grigoryan-softconstruct) [reply src/components/molecules/Tag/Tag.tsx:58]: done

File: src/components/molecules/Tag/Tag.tsx [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    text: string;
+    /**
+     * Tag type <br/>
+     * Possible values: `rest | error | warning`
+     */
+    type?: "rest" | "error" | "warning";
+    /**
+     * Disables tag
+     */
+    isDisabled?: boolean;
```
[the remark this answers is not in the document]
Comment (arthur-grigoryan-softconstruct) [reply src/components/molecules/Tag/Tag.tsx:30]: done

**Review by hamikhambardzumyan — APPROVED** [review]
Looks good to me
